content/post/2022-09-01-r-dependency/index.Rmd

```diff
@@ -0,0 +1,284 @@
+
+However, you should take this as a guideline but not add a transitive minimum R version as the minimum R version of your package unless you add a feature specific to this version.
+It is important that the minimum R version you state in your package reflects the version required for the code in your package, not in one of its dependencies.
+
+## Which approach should you choose?
+
+There is no intrinsically better choice between these approaches.
+It is more a matter of world-view and relation of the project with the users.
+
+However, you should always keep in mind that [it may be difficult for users to install or update any piece of software](https://twitter.com/jimhester_/status/1350424047893557253) and you should not force them to upgrade to very recent R versions.
+A good philosophy is to consider that users cannot upgrade their R version and that you should bump the required R version only when you are sure that all active users are already using this R version or a newer one.
+
+## Minimum R dependencies in CRAN packages
+
+Whenever you are unsure about a completely subjective choice for a R package, or any project in general, it is often good practice to look at what is done in your community.
+
+Let's start by grabbing a snapshot of the current CRAN archive:
+
+```{r, cache = TRUE}
+db <- tools::CRAN_package_db()
+```
+
+We can then isolate the R version dependency declaration:
+
+```{r}
+r_deps <- db |> 
+  # We exclude recommended pkgs as they're always shown as depending on R-devel
+  dplyr::filter(is.na(Priority) | Priority != "recommended") |> 
+  dplyr::pull(Depends) |> 
+  strsplit(split = ",") |> 
+  purrr::map(~ grep("^R ", .x, value = TRUE)) |> 
+  unlist()
+length(r_deps)
+tail(r_deps)
+```
+
+A first result of our analysis if that `r sprintf("%.0f%%", length(r_deps) / nrow(db) * 100)` of CRAN packages specify a minimum R version.
+
+As mentioned earlier, the minimum required version can be specified with a loose or strict inequality:
+
+```{r}
+(r_deps_strict <- sum(grepl("^R \\(>\\s(.+)\\)", r_deps)))
+(r_deps_loose  <- sum(grepl("^R \\(>=\\s(.+)\\)", r_deps)))
+```
+
+You can see that using a strict inequality is indeed very uncommon (`r sprintf("%.2f%%", r_deps_strict / (r_deps_strict + r_deps_loose) * 100)` of the cases).
+
+We can now continue our analysis and extract the version number itself:
+
+```{r}
+r_deps_ver <- trimws(gsub("^R \\(>=?\\s(.+)\\)", "\\1", r_deps))
+
+r_deps_ver |> 
+  table()
+```
+
+Interestingly, you can notice that some of these version numbers don't match any actual R release.
+To confirm this, we can use the [rversions package, from R-hub](/2019/04/15/rversions-1-1-0/):
+
+```{r}
+setdiff(unique(r_deps_ver), rversions::r_versions()$version)
+```
+
+We can infer the reason for the mismatch for some examples in this list:
+
+-   missing `.` between version components (e.g., `2.01`, `2.50`, `3.00`, `3.60`, `4.00`)
```

Comment:
could you replace all "e.g." with "for instance"? I was recently told it's not inclusive to use Latin abbreviations.